tests/models/test_taskinstance.py

```diff
@@ -1270,20 +1338,142 @@ def test_depends_on_past(self, dag_maker):
+                id="indirect upstream setups - all done: one skipped and one success",
+            ),
+            param(
+                "all_success",
+                0,
+                2,
+                _UpstreamTIStates(7, 0, 0, 0, 0, 7, 1, 0, 1),
+                True,
+                TaskInstanceState.UPSTREAM_FAILED,
+                False,
+                id="indirect upstream setups - all done: one success and one failed",
+            ),
+            param(
+                "all_success",
+                0,
+                2,
+                _UpstreamTIStates(7, 0, 0, 0, 0, 7, 0, 1, 1),
+                True,
+                TaskInstanceState.UPSTREAM_FAILED,
+                False,
+                id="indirect upstream setups - all done: one skipped and one failed",
+            ),
+            param(
+                "all_success",
+                0,
+                2,
+                _UpstreamTIStates(7, 0, 0, 0, 0, 7, 0, 0, 1),
+                True,
+                TaskInstanceState.UPSTREAM_FAILED,
+                False,
+                id="indirect upstream setups - one not done and one failed",
+            ),
+            param(
+                "all_success",
+                0,
+                2,
+                _UpstreamTIStates(0, 0, 1, 0, 0, 1, 2, 0, 0),
+                True,
+                TaskInstanceState.UPSTREAM_FAILED,
+                False,
+                id="indirect upstream setups - all setup success but upstream failed",
+            ),
+            param(
+                "all_success",
+                0,
+                2,
+                _UpstreamTIStates(5, 0, 0, 0, 0, 5, 2, 0, 0),
+                True,
+                None,
+                False,
+                id="indirect upstream setups - all setup success but upstream still running",
+            ),
+            param(
+                "all_success",
+                0,
+                2,
+                _UpstreamTIStates(5, 0, 0, 0, 0, 5, 0, 0, 1),
+                True,
+                TaskInstanceState.UPSTREAM_FAILED,
+                False,
+                id="indirect upstream setups - one setup failed but upstream still running",
+            ),
+            param(
+                "all_success",
+                0,
+                2,
+                _UpstreamTIStates(4, 0, 1, 0, 0, 5, 0, 2, 0),
+                True,
+                TaskInstanceState.UPSTREAM_FAILED,
+                False,
+                id="indirect upstream setups - all setup skipped but upstream failed",
+            ),
+            param(
+                "all_success",
+                0,
+                2,
+                _UpstreamTIStates(0, 7, 0, 0, 0, 7, 0, 0, 1),
+                True,
+                TaskInstanceState.UPSTREAM_FAILED,
+                False,
+                id="indirect upstream setups - one setup failed but upstream skipped",
+            ),
         ],
     )
     def test_check_task_dependencies(
```

Comment:
i think that some of these tests, where you've made substantial changes, i think we should just make a new test. 

it's very important that we're confident that we're not breaking anything here, so rather than make such significant changes to the test, i think it's preferable to add a new one alongside.  i'll try to do this right now.

Reply:
It was my first thought, but I tried to avoid duplicating the code. But yeah +1